### ISSUE TITLE
Make CupertinoTextField call onChanged when clear button tapped

### ISSUE THREAD

**Issue by mleonhard** [flutter/flutter#29345]
CupertinoTextField does not call onChanged when its clear button is tapped.  This is a problem because my app needs to know when the user has cleared the search field so it can display unconstrained suggestions (including paid ads).

Here's a minimal reproduction.  Tap the '(x)' button to clear the text box.  Note that no message is printed.  I tested debug builds on iOS Simulator and Android Emulator.
```
import 'package:flutter/cupertino.dart'
    show CupertinoApp, CupertinoPageScaffold;
import 'package:flutter/widgets.dart' show Center, runApp;
import 'package:flutter/cupertino.dart'
    show CupertinoTextField, OverlayVisibilityMode, TextEditingController;

void main() {
  runApp(
    CupertinoApp(
      home: CupertinoPageScaffold(
        child: Center(
          child: CupertinoTextField(
            controller: new TextEditingController(text: 'initial text'),
            clearButtonMode: OverlayVisibilityMode.always,
            onChanged: (String contents) {
              print('onChanged $contents');
            },
          ),
        ),
      ),
    ),
  );
}
```

$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.2.1, on Mac OS X 10.14.3 18D109, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[!] iOS toolchain - develop for iOS devices (Xcode 10.1)
[✓] Android Studio (version 3.3)
[!] IntelliJ IDEA Community Edition (version 2018.3.5)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] Connected device (2 available)

! Doctor found issues in 2 categories.

**Comment by justinmc** [comment]
Thanks for the super clear example!  I tried it and I see the same behavior that you see.

This seems like a good idea to me if no one jumps in with a reason not to.

**Comment by xster** [comment]
Seems like a sensible fix. If anyone wants to send us a PR, it should be an easy fix.

**Comment by hyjfine** [comment]
because the design of EditableText onChanged is does run not when the TextField's text is changed programmatically, via the TextField's [controller]. Typically it isn't necessary to be notified of such changes, since they're initiated by the app itself.
and CupertinoTextField cleanButton use _effectiveController.clear() to clean text. it's just changed programmatically
so two ways to fix ，redefinition the EditableText onChanged or special handle onChanged for CupertinoTextField ClearButton ？Thanks for any recommend @xster

**Comment by xster** [comment]
Fixed in #29474

**Comment by oligazar** [comment]
What about plain TextField not calling onChanged after calling TextEditingController.clear()?

**Comment by patelkrunal78** [comment]
Very Similar To [How to add clear button to TextField Widget](https://icetutor.com/question/how-to-add-clear-button-to-textfield-widget)

**Comment by milanpatel18** [comment]
Already Solved At #9254
